Assets/Scripts/Enemy/EnemyController.cs

```diff
@@ -1,21 +1,28 @@
 using System.Collections;
 using System.Collections.Generic;
+using Unity.VisualScripting;
 using UnityEngine;
 
-public abstract class EnemyController : CharacterController
+public abstract class EnemyController : CharController
 {
-    public float enemyMoveSpeed = 3f;
+    protected float enemyMoveSpeed = 3f;
+    protected GameObject refToPlayer;
```

Comment:
An attribute declared in a class should be instantiated in the class unless it's declared as abstract. You can never guarantee `refToPlayer` won't be null since you rely on the child class instantiating this variable. 

Furthermore, you instantiate attribute in all child classes in the same way, meaning you can also do it here. If you expect this attribute will be instantiated differently depending on the child class, you should declare the attribute as `abstract` to force the children to handle the attribute.